datafusion/core/src/physical_plan/tree_node/rewritable.rs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Tree node rewritable implementations
+
+use crate::physical_plan::tree_node::TreeNodeRewritable;
+use crate::physical_plan::{with_new_children_if_necessary, ExecutionPlan};
+use datafusion_common::Result;
+use std::sync::Arc;
+
+impl TreeNodeRewritable for Arc<dyn ExecutionPlan> {
```

Comment:
Is there any reason for implementing `TreeNodeRewritable` trait in another file. I guess, you could have implement this trait in the file where `ExecutionPlan` is implemented. I am not familiar with these pattern. I am asking to understand better.

Reply:
If the Orphan rule is relaxed in the future, the same trait of `TreeNodeRewritable` can be used for the `PhysicalExpr`. The traits defined in the tree_node file are common abstractions. They are not just for `ExecutionPlan`

Reply:
I understand why `TreeNodeRewritable` trait is defined in `datafusion/core/src/physical_plan/tree_node/mod.rs`. What I mean was that `Arc<dyn ExecutionPlan>` implements `TreeNodeRewritable` trait. I guess, we could do this implementation in `datafusion/core/src/physical_plan/mod.rs`. However, it is just a stylistic issue. I just wondered the reason why you did it in  `datafusion/core/src/physical_plan/tree_node/rewritable.rs`